File: dual-tales-backend/src/main/java/com/example/dual_tales/api/story/dto/StoryDraftResponseDto.java
package com.example.dual_tales.api.story.dto;

import com.example.dual_tales.domain.story_draft.StoryDraft;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class StoryDraftResponseDto {
    private Long draftId;
    private String question_ko;
    private String question_foreign;
    private int currentStep;
    private boolean isFinal;

    public static StoryDraftResponseDto from(StoryDraft draft,String qKo, String qForeign, boolean isFinal) {
        return StoryDraftResponseDto.builder()
                .draftId(draft.getId())
                .question_ko(qKo)
                .question_foreign(qForeign)
                .currentStep(draft.getCurrentStep())
                .isFinal(isFinal)
                .build();
    }
}

File: dual-tales-android/app/src/main/java/com/example/dualtales/network/dto/StoryDraftResponseDto.kt
package com.example.dualtales.network.dto

import com.google.gson.annotations.SerializedName

data class StoryDraftResponseDto(
    val draftId: Long,
    val question_ko: String,
    val question_foreign: String,
    val currentStep: Int,
    @SerializedName("final") val isFinal: Boolean
)
